packages/gatsby/src/utils/webpack-utils.js

```diff
@@ -609,6 +610,11 @@ module.exports = async ({
     }
   ) => new OptimizeCssAssetsPlugin(options)
 
+  plugins.fastRefresh = () =>
+    new ReactRefreshWebpackPlugin({
+      disableRefreshCheck: true,
```

Comment:
explore why this is necessary because it feels weird to me that it needs to be disabled. Add a comment if this is right.